src/requests/GroupsApi.ts

```diff
@@ -0,0 +1,1246 @@
+
+    constructor(baseClass: VRCWrapper) {
+        super(baseClass);
+        this.baseClass = baseClass;
+    }
+
+    // Creates a Group and returns a Group object. Requires VRC+ Subscription.
+    /**
+     * **YOU MUST HAVE VRC+ SUBSCRIPTION TO USE THIS METHOD!**
+     * 
+     * @param {VRCAPI.Generics.dataKeysCreateGroupRequest} parameters  An object with the following:
+     * 
+     * @param name `string` - The name of the group. Must be between 3 and 64 characters. **[REQUIRED]**.
+     * @param shortCode `string` - The short code of the group. Must be between 3 and 6 characters. **[REQUIRED]**.
+     * @param roleTemplate **ENUM** `VRCAPI.Groups.Models.GroupRoleTemplate` - The role template of the group. [DEFAULT: `0` - `Default`] **[REQUIRED]**.
+     * @param description `string` - The description of the group. Must be between 0 and 250 characters. *[OPTIONAL]*.
+     * @param joinState **ENUM** `VRCAPI.Groups.Models.GroupJoinState` - The join state of the group. [DEFAULT: `0` - `Open`] *[OPTIONAL]*.
+     * @param iconId `string` - The icon id of the group. Must be a valid image id. *[OPTIONAL]*.
+     * @param bannerId `string` - The banner id of the group. Must be a valid image id. *[OPTIONAL]*.
+     * @param privacy **ENUM** `VRCAPI.Groups.Models.GroupPrivacy` - The privacy of the group. [DEFAULT: `0` - `Default`] *[OPTIONAL]*
+     * @returns {Promise<VRCAPI.Groups.Models.Group>} - Returns a Group object Promise. 
+     */
+    public async createGroup({
+        name,
+        shortCode,
+        roleTemplate,
+        description,
+        joinState,
+        iconId,
+        bannerId,
+        privacy,
+    }: VRCAPI.Groups.Requests.CreateGroupRequest): Promise<VRCAPI.Groups.Models.Group> {
+
+        if (name.length < 3 || shortCode.length > 64) return Promise.reject(new Error('Name must be between 3 and 64 characters!'));
+        if (shortCode.length < 3 || shortCode.length > 6) return Promise.reject(new Error('ShortCode must be between 3 and 64 characters!'));
+        if (!Object.values(GroupRoleTemplate).includes(roleTemplate)) return Promise.reject(new Error('RoleTemplate must be one of the following: 0 - Default, 1 - ManagedFree, 2 - ManagedInvite, 3 - ManagedRequest!'));
+
+        const body: VRCAPI.Generics.dataKeysCreateGroupRequest = {
+            name,
+            shortCode,
+            roleTemplate,
+        };
+
+        if (description) {
+            if (description.length > 250 || description.length < 0) return Promise.reject(new Error('Description must be between 0 and 250 characters!'));
+            body.description = description;
+        }
+
+        if (joinState) {
+            if (!Object.values(GroupJoinState).includes(joinState)) {
+                console.warn('JoinState must be one of the following: 0 - Closed, 1 - Invite, 2 - Request, 3 - Open! Parameter ommited! Defaulted to "0 - Open"');
+            } else {
+                body.joinState = joinState;
+            }
```

Comment:
You can probably do this with a ENUM instead of verifying in code.

Reply:
I'm not sure I understand because GroupJoinState is an enum already 🤔

Reply:
Your GroupJoinState enum is: 
```typescript
export enum GroupJoinState {
    Open = 'open',
    Closed = 'closed',
    Invite = 'invite',
    Request = 'request',
}
```

But here with the way you're using it, it should be:
```typescript
export enum GroupJoinState {
    Closed = 0,
    Invite = 1,
    Request = 2,
    Open = 3,
}
```

Reply:
That's why i told you, you should review what [Enums do in Typescript](https://www.typescriptlang.org/docs/handbook/enums.html)